### PR TITLE
Improve GitHub Actions

### PR DESCRIPTION
**Pull request by boramalper**
- Skip vHive tests on changes to `./function-composition/**`
- Build chained-functions-eventing automatically
- function-composition CI/CD now waits for successful tests before pushing to Docker Hub

Signed-off-by: Bora M. Alper <bora@boramalper.org>

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -6,11 +6,13 @@ on:
     paths-ignore:
     - 'docs/**'
     - '**.md'
+    - 'function-images/**'
```
Comment (ustiugov) [inline .github/workflows/build.yml:9]: this workflow can run, it's not heavy and it's not self-hosted

File: .github/workflows/function-composition.yml [hunk 2]
```diff
@@ -1,19 +1,108 @@
...
 env:
   GOOS: linux
   GO111MODULE: on
+  VHIVE_GOLANG_VERSION: golang:1.16.4-alpine3.13
 
 jobs:
-  docker-compose:
-    name: Serving uBench (docker-compose)
+  push-golang:
+    name: Push golang
```
Comment (ustiugov) [inline .github/workflows/function-composition.yml:25]:
```suggestion
    name: Push Golang build image
```

**Comment by boramalper** [comment]
Addresses #282.

File: .github/workflows/image_build.yml [hunk 3]
```diff
@@ -4,69 +4,17 @@ on:
   workflow_dispatch:
```
Comment (ustiugov) [inline .github/workflows/image_build.yml:4]: remove this file

**Review by ustiugov — CHANGES_REQUESTED** [review]
- [ ] separate build & push workflow from the composition tests. 
- [ ] make composition tests workflow dependent on the build workflow

**Review by ustiugov — APPROVED** [review]

**Pull request merged**